app/controllers/niches_controller.rb

```diff
@@ -0,0 +1,71 @@
+class NichesController < ApplicationController
+  include ApplicationConcern
+
+  before_action :set_niche, only: %i[show edit update destroy]
+
+  # GET /locations
+  # GET /locations.json
+  def index
+    @niches = current_user.niches.reject(&:parent)
+  end
+
+  # GET /locations/1
+  # GET /locations/1.json
+  def show; end
+
+  # GET /locations/new
+  def new
+    @niche = current_user.niches.create
```

Comment:
it's kind of dangerous to create objects on form opening, do you remove them if user doesn't save form?